File: tests/encode.mjs
import { argv } from "node:process";

import {
  OPS,
  encodeOps,
  opsBitWidths,
  UNIQUE_SEEDS,
  encodeSeeds,
  seedsBitWidths,
} from "./ops_andSeeds_codecs.mjs";

for (const [set, encoder, bitWidths] of [
  [OPS, encodeOps, opsBitWidths],
  [UNIQUE_SEEDS, encodeSeeds, seedsBitWidths],
]) {
  const symbols = argv.slice(2).filter((arg) => set.has(arg));
  if (symbols.length === 0) {
    continue;
  }

  const encodedHexStrs = encoder(symbols.values(), bitWidths).map((x) =>
    Number(x).toString(16).padStart(2, "0"),
  );

  console.log(encodedHexStrs.reduce((s1, s2) => `${s1}${s2}`));
}

File: tests/encode.py
import sys

import sub_byte


for symbols, encoder_func in [(core.OPS, encoder.encode_ops),
                              (core.ALL_SEEDS, encoder.encode_seeds),
                             ]:
    parsed_args = [arg for arg in sys.argv[1:] if arg in set(symbols)]
    print(bytes(encoder_func(parsed_args)).hex())